content/docs/user-guide/gems/reference/kythera-ai/spatial-query-system.md

```diff
@@ -0,0 +1,723 @@
+
+            
+
+When an agent needs to choose a position to move to, it can do so by executing a Spatial Query which will return the best position for a given set of criteria. The query itself is described in a specialized query language, comprising:
+
+*   **Generators**, which produce a set of positions to consider
+    
+*   **Conditions**, which test each position and exclude inappropriate ones
+    
+*   **Scores**, which weight and rank the appropriate positions in order to find the best candidate
+    
+
+A query is constructed once, then executed any number of times by the Spatial Query System. The system handles queuing, time-slicing and asynchronous processing of multiple queries over a number of frames. When a query is completed, it uses the Kythera signal mechanism to communicate the result back to the original caller.
+
+Spatial queries are specified in text files, which are parsed and interpreted at run-time. This allows rapid iteration, in particular for high-level agent-based queries, as the files can be reloaded while the game is running.
+        
+
+## Writing spatial queries
+
+Each level in a game can include a file called `SpatialQueries.sqs` which defines a collection of Spatial Queries. Queries common to multiple levels can be placed in the global Scripts folder. Whenever a level is loaded, all queries are first unloaded and then the global queries are loaded in. The level-specific queries come next and will overwrite any global queries with the same name.
+
+There is a console command, `kyt_ReloadSpatialQueries`, which will reload the queries in the file on-demand, allowing rapid iteration during development.
+In addition to the queries written in C++ as shown above, queries can also be provided in a file. These files will be parsed by the Spatial Query System, and used to construct spatial queries. The syntax for these queries is the same as for C++ queries.
+
+
+### Query structure
+
+A spatial query consists of four blocks - `Generators`, `Conditions`, `Scores` and `Fallbacks`. When the query is executed:
+
+*   The Generators are run in parallel (via time-slicing) and each returns one or more candidate positions or entities (such as cover points) as output
+    
+*   Once all Generators have completed, this set of candidate positions is passed through each Condition, in order, and any that fail are immediately dropped from further processing
+    
+*   Each remaining candidate position is now assigned a score by evaluating each Score criterion. These consist of a weight and a criterion that produces a score. Each criterion score is multiplied by its weight; the sum of all these values is the total score assigned to the position.
```

Comment:
```suggestion
*   Each remaining candidate position is assigned a score by evaluating each individual Score criterion. These consist of a weight and a criterion that produces a score. Each criterion score is multiplied by its weight; the sum of all these values is the total score assigned to the position.
```

Not sure what's up with the stray caps of `Score` here - could be a typo or an un-bolded term?